### PR TITLE
(DOCKER) allows to use jbarlow83's official docker image of OCRmyPDF, i.e. use a newer version

### PR DESCRIPTION
**Pull request by totti4ever**
- if `/var/run/docker.sock` is found in the docker-container, this feature is activated - if not, nothing changes
- usage:
  - mount bind `docker.sock` from host by using `-v` or `volumes:`
  - set environment variable when a certain version of the used docker image is preferred: `OCRMYPDF_VERSION=v11.3.0` or `OCRMYPDF_VERSION=latest`, default (if not set) is `v11.2.1`

addresses #387

### CODE REVIEW

File: docker/joex-entrypoint.sh [hunk 1]
```diff
@@ -3,4 +3,15 @@
 echo "Starting unoconv listener"
 unoconv -l &
 
+# replace own ocrmypdf with official dockerfile, i.e. newer version
+if [ -S "/var/run/docker.sock" ]; then
+  echo "Found 'docker.sock': Installing Docker and redirecting 'ocrmypdf' command to official dockerfile by jbarlow83"
+  apk --no-cache add docker
+  docker pull -q jbarlow83/ocrmypdf:$OCRMYPDF_VERSION
+  function ocrmypdf () {
```
Comment (eikek) [inline docker/joex-entrypoint.sh:11]: This is really interesting. this function is enough to declare such that it is in the `PATH` and docspell can call it?

Comment (eikek) [reply docker/joex-entrypoint.sh:11]: Ah it is in the environment of the parent process right? It's not necessary to `export` it? (sorry, this is not a critique I'm just curious)

**Comment by eikek** [comment]
Could you maybe show how to change the `docker-compose.yml` (as an example) to use this new feature? I'll add it somewhere to the docs as the recommended way?

File: docker/joex-entrypoint.sh (hunk 1 above)
Comment (totti4ever) [reply docker/joex-entrypoint.sh]: I never take your questions as critique, plus usually they are totally important!

As in this case - i only tested in interactively, and did not put enough attention to it when testing in the final environment.

In the end, I had to move from alias to bash function now to script in `/usr/local/bin` which comes before `/usr/bin` where the original ocrmypdf executable is

**Comment by totti4ever** [comment]
> Could you maybe show how to change the `docker-compose.yml` (as an example) to use this new feature? I'll add it somewhere to the docs as the recommended way?

What you actually need to do is to mount bind two things into the joex container:
```
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - /tmp/docspell-convert:/tmp/docspell-convert
```

Talking about the docs:
What are you using, is it code-based? Could it be build from the or another git-repo where others could provide PRs?

**Comment by eikek** [comment]
Docs are in the same repo, in folder [`website`](https://github.com/eikek/docspell/tree/master/website).

**Pull request merged**